src/psyclone/psyGen.py

```diff
@@ -460,14 +468,16 @@ def __init__(self, alg_invocation, idx, schedule_class, invokes,
                 alg_invocation.kcalls[0].ktype.name
         else:
             # use the position of the invoke
-            self._name = "invoke_"+str(idx)
+            self._name = "invoke_" + str(idx)
 
         if not reserved_names:
             reserved_names = []
-        reserved_names.append(self._name)
 
         # create the schedule
-        self._schedule = schedule_class(alg_invocation.kcalls, reserved_names)
+        self._schedule = schedule_class(self._name, alg_invocation.kcalls,
+                                        reserved_names)
+        if self.invokes:
+            self.invokes.psy.psy_container.addchild(self._schedule)
```

Comment:
Possibly, but since this still just goes towards #1010 and for now I don't want to interfere with the existing code too much I propose we revisit this idea in a latter PR.

Also we have to think how to do the Fortran-to-C 2-level PSy-layer which will have 2 containers (but maybe just appear in a later stage in the C visitor?)

I added a TODO in the constructor with this idea.